app/controllers/Support.scala

```diff
@@ -45,30 +60,39 @@ class Support(
      }
   }
 
-  def validateImageDimensions(image: File, requiredWidth: Option[Long], requiredHeight: Option[Long]): Boolean = {
-    (requiredWidth, requiredHeight) match {
+  def validateImageDimensions(image: File, requiredWidth: Option[Long], requiredHeight: Option[Long]): Either[String, File] = {
+    val dimensionsAreValid = (requiredWidth, requiredHeight) match {
       case (None, None) => true
       case (Some(width), None) => width >= ImageMetadataService.imageWidth(image)
       case (None, Some(height)) => height >= ImageMetadataService.imageHeight(image)
       case (Some(width), Some(height)) => width >= ImageMetadataService.imageWidth(image) && height >= ImageMetadataService.imageHeight(image)
     }
+    dimensionsAreValid match {
+      case true => Right(image)
+      case false => Left(s"Image must have dimensions w:${requiredWidth.getOrElse("Not Specified")}px h:${requiredHeight.getOrElse("Not Specified")}px")
+    }
   }
 
   def uploadLogo(filename: String) = APIAuthAction(parse.temporaryFile) { req =>
     val picture = req.body
     val requiredWidth = req.getQueryString("width").map(_.toLong)
     val requiredHeight = req.getQueryString("height").map(_.toLong)
 
-    validateImageDimensions(picture.file, requiredWidth, requiredHeight) match {
-      case true => {
+    val imageValidationResult = for {
+      fileWithValidExtension <- checkFileExtension(picture.path.toFile, filename)
+      image <- validateImageDimensions(fileWithValidExtension, requiredWidth, requiredHeight)
+    } yield image
+
+    imageValidationResult match {
```

Comment:
it won't let me do a suggestion for the whole block, but I recommend using `.fold` on the Either to replace this outer pattern match, to avoid confusion with the inner pattern match on a different Either.

Reply:
Good suggestion, have implemented now.